GCTC-NTGC/gc-digital-talent
# 6163 - implement webpack text compression

## 👋 Introduction

Closes #6163

Observing the main suggestion of #6071
I attempted implementing compression to test how it works locally. 

Think it is... promising?

## 🕵️ Details

Googled to get a feel for something to try, and implemented things that I found. Like from reading this
https://stackoverflow.com/questions/57504666/enable-text-compression-using-react-webpack-and-apache

Uses
https://github.com/webpack-contrib/compression-webpack-plugin

## 🧪 Testing

1. docker down then up
2. run build
3. observe the site

## 📸 Screenshot

Comparison of results for the homepage

![image](https://user-images.githubusercontent.com/40485260/229205675-e68e0096-3c6f-4210-926e-737ac6b74b2e.png)

Lighthouse run for the homepage (contrast to https://github.com/GCTC-NTGC/gc-digital-talent/issues/6071#issuecomment-1492068357)

![image](https://user-images.githubusercontent.com/40485260/229205716-21c77aa6-ffb8-4130-aba4-237eee808548.png)
